src/client/components/sms-fraud/SubmitCodeForm.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from 'classnames';
+import { FunctionComponent, useState } from 'react';
+import { useMutation } from 'react-query';
+import { SubmitCodeResponse, SubmitCodePayload } from '../../../pages/api/sms-fraud/submit-code';
+import { TEST_IDS } from '../../testIDs';
+import { useVisitorData } from '@fingerprintjs/fingerprintjs-pro-react';
+import { SendMessageButton } from './SendSMSMessageButton';
+import { Alert } from '../common/Alert/Alert';
+import Button from '../common/Button/Button';
+import styles from './smsVerificationFraud.module.scss';
+import formStyles from '../../../styles/forms.module.scss';
+import { SendMessageMutation } from '../../../pages/sms-fraud';
+
+export const useSubmitCode = (params?: { onSuccess?: () => void }) => {
+  const { getData } = useVisitorData(
+    { ignoreCache: true },
+    {
+      immediate: false,
+    },
+  );
+  return useMutation<SubmitCodeResponse, Error, { phoneNumber: string; code: string }>({
+    mutationKey: ['submitCode'],
+    mutationFn: async ({ code, phoneNumber }) => {
+      const { requestId } = await getData();
+      const response = await fetch(`/api/sms-fraud/submit-code`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          code: Number(code),
+          phoneNumber,
+          requestId,
+        } satisfies SubmitCodePayload),
+      });
+      if (response.status < 500) {
+        return await response.json();
+      } else {
+        throw new Error('Failed to submit code: ' + response.statusText);
+      }
```

Comment:
Let's unwrap the if/else

Reply:
Fixed, thanks :) https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/pull/135/commits/218fc51971d7d095ee7c50c87956bc6e8afcbcfb